src/hooks/getAssetCapUsage.tsx

```diff
@@ -0,0 +1,128 @@
+import { valueToBigNumber } from '@aave/math-utils';
+import { SupplyCapWarning } from 'src/components/transactions/Warnings/SupplyCapWarning';
+import { BorrowCapWarning } from 'src/components/transactions/Warnings/BorrowCapWarning';
+import { DebtCeilingWarning } from 'src/components/transactions/Warnings/DebtCeilingWarning';
+import { ComputedReserveData } from './app-data-provider/useAppDataProvider';
+import { SupplyCapTooltip } from 'src/components/infoTooltips/SupplyCapTooltip';
+import { BorrowCapTooltip } from 'src/components/infoTooltips/BorrowCapTooltip';
+import { DebtCeilingTooltip } from 'src/components/infoTooltips/DebtCeilingTooltip';
+
+type WarningDisplayProps = {
+  supplyCap?: AssetCapData;
+  borrowCap?: AssetCapData;
+  debtCeiling?: AssetCapData;
+  icon?: boolean;
+  useDefaultTooltip?: boolean;
+};
+
+export type AssetCapData = {
+  percentUsed: number;
+  isMaxed: boolean;
+};
+
+export type AssetCapHookData = AssetCapData & {
+  determineWarningDisplay: (props: WarningDisplayProps) => JSX.Element | null;
+  determineTooltipDisplay: (props: WarningDisplayProps) => JSX.Element | null;
+};
+
+export type AssetCapUsageData = {
+  supplyCap: AssetCapHookData;
+  borrowCap: AssetCapHookData;
+  debtCeiling: AssetCapHookData;
+};
+
+type AssetLikeObject =
+  | ComputedReserveData
+  | (unknown & {
+      // Just whitelisting the fields used in this 'hook'
+      totalLiquidity: string;
+      supplyCap: string;
+      totalDebt: string;
+      borrowCap: string;
+      isolationModeTotalDebt: string;
+      debtCeiling: string;
+    })
+  | undefined;
+
+const getAssetCapUsage = (asset: AssetLikeObject): AssetCapUsageData => {
+  /*
+    Supply Cap Data
+    % of totalLiquidity / supplyCap
+    Set as zero if either is zero
+  */
+  let supplyCapUsage: number = asset
+    ? valueToBigNumber(asset.totalLiquidity).dividedBy(asset.supplyCap).toNumber() * 100
+    : 0;
+  supplyCapUsage = supplyCapUsage === Infinity ? 0 : supplyCapUsage;
+  const supplyCapReached = supplyCapUsage !== Infinity && supplyCapUsage >= 99.99;
```

Comment:
I believe the `dividedBy` result can actually come back as `Infinity` anytime `asset.supplyCap` is zero in this case.  It would be useful to abstract the `x / 0` case within the utility function itself though so this logic is cleaner and it's centralized to that particular utility.

Reply:
Yeah I had the same thought about creating utility function for that.

The result coming back as `Infinity` makes sense, but on line 56 you handle that case. So on line 57, `suppleCapUsage !== Infinity` should always be true.